include/arbor/mc_segment.hpp

```diff
@@ -54,8 +54,13 @@ struct mechanism_desc {
     }
 
     double get(const std::string& key) const {
-        auto optv = util::value_by_key(param_, key);
-        return optv? *optv: throw std::out_of_range("no field "+key+" set");
+        auto i = param_.find(key);
+        if (i==param_.end()) {
```

Comment:
I have a memo from a C++ developer in the future, it reads simply "thank you"!

Reply:
though why the `else`?

Reply:
No semantic reason; just thought it aided clarity. Would happily remove.